### ISSUE TITLE
Refactor code to remove tenacity dependency 

### ISSUE THREAD

**Issue by ganeshrn** [ansible-community/ansible-lint#1782]
##### Summary

Refactor code to remove tenacity dependency and maybe move the retry logic to [ansible-compat](https://github.com/ansible-community/ansible-compat)

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible-lint --version

```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behaviour

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behaviour

Please give some details of what is actually happening.
Include a [minimum complete verifiable example] with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
